### PR TITLE
Adding performance tests

### PR DESCRIPTION
**Pull request by Sajjon**

### CODE REVIEW

File: BitcoinKit/BitcoinKitPrivate.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ NS_ASSUME_NONNULL_BEGIN
 @end
 
 @interface _Key : NSObject
-
++ (NSData *)curveOrderMinus:(NSInteger)smallNumber;
```
Comment (Sajjon) [inline BitcoinKit/BitcoinKitPrivate.h]: @kishikawakatsumi we probably do not want this in the `BitcoinKitPrivate.h` file? But I was unsure how to create an Objective C `.m`/`.h` file tuple in the Tests and refer to OpenSSL. I created an `.m` file, a bridging header and added Library Search paths to include openssl but didn't get it to work.

File: BitcoinKit/BitcoinKitPrivate.m [hunk 2]
```diff
@@ -45,6 +47,32 @@ + (NSData *)hmacsha512:(NSData *)data key:(NSData *)key {
 
 @implementation _Key
 
++ (NSData *)curveOrderMinus:(NSInteger)smallNumber {
+    Byte *byteData = (Byte*)malloc(4);
```
Comment (Sajjon) [inline BitcoinKit/BitcoinKitPrivate.m]: `NSInteger -> NSData -> BIGNUM` conversion

File: BitcoinKit/Script.swift [hunk 3]
```diff
@@ -10,16 +10,24 @@ import Foundation
...
+    private static let OP_DUP: Data = 0x76
+    private static let OP_HASH160: Data = 0xa9
+    private static let OP_0: Data = 0x14
+    private static let OP_EQUALVERIFY: Data = 0x88
+    private static let OP_CHECKSIG: Data = 0xac
 
     // Standard Transaction to Bitcoin address (pay-to-pubkey-hash)
     // scriptPubKey: OP_DUP OP_HASH160 OP_0 <pubKeyHash> OP_EQUALVERIFY OP_CHECKSIG
     public static func buildPublicKeyHashOut(pubKeyHash: Data) -> Data {
-        return Data() + OP_DUP + OP_HASH160 + OP_0 + pubKeyHash + OP_EQUALVERIFY + OP_CHECKSIG
```
Comment (Sajjon) [inline BitcoinKit/Script.swift]: actually this previous onliner made Xcode 9.4.1 fail to compile due to too complex statement (running on a top of the line Macbook Pro late 2016). But I think my changes are quite nice using `reduce` and `Data`.

File: BitcoinKitTests/PerformanceTests/MultiplicationTests.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+}
+extension Int: ByteConvertible {}
+extension Int {
+    func toData() -> Data {
+        return Data(to32Bytes())
+    }
+}
+
+private func privateKeyWifFromInt(_ int: Int, network: Network = .mainnet) -> String {
```
Comment (Sajjon) [inline BitcoinKitTests/PerformanceTests/MultiplicationTests.swift]: this was the only was I was able to perform point multiplication with low numbers `[1, 2, 3, 4,..., 999, 1000]` with the Curve order.

File: BitcoinKit/BitcoinKitPrivate.h (hunk 1 above)
Comment (usatie) [reply BitcoinKit/BitcoinKitPrivate.h]: Thanks.
I'm not sure about how to include C file in the Tests.
But `curveOrderMinus` is not suitable to be included in `BitcoinKitPrivate.h `.

**Comment by usatie** [comment]
Hi @Sajjon , thanks for dropping us the PR!

Can you make the PR that only contains the Performance tests?

**Pull request closed without merging**